features/guidance/staggered-intersections.feature

```diff
@@ -0,0 +1,72 @@
+@routing  @guidance @staggered-intersections
+Feature: Staggered Intersections
+
+    Background:
+        Given the profile "car"
+        Given a grid size of 1 meters
+        # Note the one meter grid size: staggered intersections make zig-zags of a couple of meters only
+
+    # https://www.openstreetmap.org/#map=19/39.26022/-84.25144
+    Scenario: Staggered Intersection: Oak St, Cedar Dr
+        Given the node map
+            |   |   | j |   |   |
+            | a | b | c |   |   |
+            |   |   | d |   |   |
+            |   |   | e | f | g |
+            |   |   | h |   |   |
+            |   |   | i |   |   |
+
+        And the ways
+            | nodes  | highway     | name     |
+            | abc    | residential | Oak St   |
+            | efg    | residential | Oak St   |
```

Comment:
Yes, this hooks into the pre-existing name-oscillation logic.

Added a test for this: https://github.com/Project-OSRM/osrm-backend/pull/2738/commits/b83151a1939fc5374fc18501864dde344f6ea698#diff-3e526a8f2e6b88dc336b8b12f7ca5f52R74